Tests/IdentifyInterceptorTests.m

```diff
@@ -276,4 +276,92 @@ - (void)testMultipleInterceptedIdentifyIsAppliedToNextActiveEvent {
     XCTAssertTrue([userProperties3[@"set-key-2"] isEqualToString:@"set-value-b"]);
     XCTAssertTrue([userProperties3[@"set-key-3"] isEqualToString:@"set-value-d"]);
 }
+
+- (void)testNullValuesInIdentifySetAreIgnoredOnActiveIdentify {
+    // intercept identify 1
+    AMPIdentify *identify1 = [AMPIdentify.identify set:@"set-key" value:@"set-value-a"];
+    [identify1 set:@"set-key-2" value:@"set-value-b"];
+    NSMutableDictionary *event1 = [self getIdentifyEvent:identify1];
+
+    event1 = [self->_identifyInterceptor intercept:event1];
+
+    XCTAssertNil(event1);
+    XCTAssertEqual(self->_dbHelper.getLastSequenceNumber, 1);
+    XCTAssertEqual(self->_dbHelper.getInterceptedIdentifyCount, 1);
+
+    // intercept identify 2
+    AMPIdentify *identify2 = [AMPIdentify.identify set:@"set-key" value:nil];
+    [identify2 set:@"set-key-2" value:@"set-value-c"];
+    [identify2 set:@"set-key-3" value:nil];
+    NSMutableDictionary *event2 = [self getIdentifyEvent:identify2];
+
+    event2 = [self->_identifyInterceptor intercept:event2];
+
+    XCTAssertNil(event2);
+    XCTAssertEqual(self->_dbHelper.getLastSequenceNumber, 2);
+    XCTAssertEqual(self->_dbHelper.getInterceptedIdentifyCount, 2);
+
+    // active identify
+    AMPIdentify *identify3 = [AMPIdentify.identify add:@"add-key" value:[NSNumber numberWithInt:1]];
+    NSMutableDictionary *event3 = [self getIdentifyEvent:identify3];
```

Comment:
I'm not familiar with the language so correct me if I'm wrong, but is this collapsing an ident with the `$add` key?

Does that mean we're collapsing idents regardless of the keys? There is an order of operations for user property operations, so I'm concerned this won't actually be correct. i.e. Sending 2 identify events, one with a set and one with a remove vs a single event with both a set and remove, the behavior could be incorrect if the remove is processed first

Reply:
For reference, the server side only collapses idents with `set` and otherwise just leaves the event, and "resets" the merged user properties in memory.

Reply:
Or is this `$add` event included to "tell" the interceptor the previous chain of idents is done?

Reply:
Currently an Identify event with non-set actions will cause intercepted set-actions to be collapsed onto it, then sent as a single Identify.

If this is incorrect we will need to update the handling in all SDKs. Let me know.

cc @qingzhuozhen

Reply:
See lines 314-323 in this file for the structure of the merged identify event.

Reply:
sync'd up offline, this behavior is half correct.

Reply:
I made a new ticket for this change - https://amplitude.atlassian.net/browse/AMP-73588